JoshuaOndieki/ridemyway-api
# [Delivers #158791391] API v2 Signup Endpoint

**What does this PR do?**
Adds signup endpoint to the API

**Description of Task to be completed**
Adding a signup endpoint to allow drivers | riders to sign up

POST `/api/v2/auth/signup`

**Tasks achieved in this PR**
- [x] POST `/api/v2/auth/signup`


**How should one test this endpoint?**
- Clone this repo using `git clone https://github.com/JoshuaOndieki/ridemyway-api.git`
- Check the ReadMe for instructions on setting up the project. 
- `git checkout ft-signup-158791391` 
- Run `nosetests ridemyway/tests/test_v2/test_auth/test_signup.py` to see the signup tests pass. Alternatively, run `pytest ridemyway/tests/test_v2/test_auth/test_signup.py`

**But why is the build failing :confused: ?**
The build is also testing endpoints which are yet to be implemented. Bear with me while I implement them.

**How can you help?**
Review my work and let me know any semantic, technical or syntax/standards errors I may have.
I will greatly appreciate any feature/change suggestions too :smiley: 

**Resources & Libs**
[This Pivotal Tracker board](https://www.pivotaltracker.com/n/projects/2179581) was used to plan and manage the tasks.

**Relevant PIvotal Tracker Stories?**
[#158791391](https://www.pivotaltracker.com/story/show/158791391)
